OneCommunityGlobal/HGNRest
# Diego add conventional commits with husky

<img width="736" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/26840798/88b75186-cf15-470c-bf84-cd8a6b94ce2f">

# Description
Adds a bash script to enforce conventional commits. Conventional commits are a recommended way to write commit messages which will be later use to for creating commit history that is readable and that can self document the source code using Changelogs. This can be further automated with through a custom action in github workflows.

## Related PRS:
N/A

## Main changes explained:
- Adds a bash script file in .husky to bind to the commit-msg git hook to enforce commit messages complying with conventional commits specification.

## How to test:
1. check into current branch
2. do `npm install`
3. if you are working on Mac or Linux, run the following command: `npm run prepare` (this will make .husky files executable)
4. create a new branch of you own for testing this PR
5. perform any changes in file and add a commit message that doesn't comply with the conventional commits specification (it should fail to commit your staged files)
6. write a commit message that does comply with conventional commits specification (it should pass)
7. watch the attached video to understand how properly follow these directions.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/26840798/52bd7afc-7d07-4f51-889b-26517b74ad68

## Note:
This is approach to write our commit messages should be enforced in order to better document our code and have a better commit history which can be easily tracked. In addition, automated documenting tools like changelogs will leverage this conventional commits to further document the code in a MarkDown file.
